src/Admin/SiteHealth.php

```diff
@@ -223,6 +294,126 @@ static function ( $available_service ) {
+			'description' => $description,
+			'test'        => self::TEST_PAGE_CACHING,
+			'status'      => $status,
+			'label'       => esc_html( $label ),
+			'actions'     => sprintf(
+				'<p><a href="%1$s" target="_blank" rel="noopener noreferrer">%2$s <span class="screen-reader-text">%3$s</span><span aria-hidden="true" class="dashicons dashicons-external"></span></a></p>',
+				esc_url( 'https://amp-wp.org/documentation/getting-started/amp-site-setup/page-caching-with-amp-and-wordpress/' ),
+				esc_html__( 'Learn more about page caching', 'amp' ),
+				/* translators: The accessibility text. */
+				esc_html__( '(opens in a new tab)', 'amp' )
+			),
+		];
+	}
+
+	/**
+	 * Check if site has page cache enable or not.
+	 *
+	 * @return array {
+	 *     @type bool $server_caching True if site has page cache. Otherwise False.
+	 *     @type bool $client_caching True if site has page cache and passed appropriate header.
+	 * }
+	 */
+	private function get_page_cache_status() {
+
+		$has_page_cache = [
+			'server_caching' => false,
+			'client_caching' => false,
+		];
+		$response_list  = [];
+		$request_args   = [
+			'headers' => [
+				'X-AMP-Request-Random-Number' => true,
+				'Cache-Control'               => 'max-age=' . ( YEAR_IN_SECONDS + MONTH_IN_SECONDS ),
+			],
+		];
+
+		for ( $i = 1; $i <= 3; $i++ ) {
+			$http_response        = wp_remote_get( home_url(), $request_args );
+			$http_response_header = wp_remote_retrieve_headers( $http_response );
+			$http_response_header = ( ! empty( $http_response_header ) ) ? $http_response_header : [];
+			$http_response_header = ( is_array( $http_response_header ) ) ? $http_response_header : $http_response_header->getAll();
+
+			$response = false;
+
+			// If any of header tag exist. and x-amp-random-number doesn't consider page cache exists.
+			if (
+				! empty( $http_response_header['cache-control'] ) ||
+				! empty( $http_response_header['expires'] ) ||
+				! empty( $http_response_header['last-modified'] ) ||
+				! empty( $http_response_header['etag'] )
+			) {
+				$has_page_cache['client_caching'] = true;
+			}
+
+			if ( ! empty( $http_response_header['x-amp-random-number'] ) ) {
+				$response = $http_response_header['x-amp-random-number'];
+			} else {
+				// If age tag have positive number then consider page cache exists.
+				if ( ! empty( $http_response_header['age'] ) && 0 < (int) $http_response_header['age'] ) {
+					$response = true;
+				}
+			}
+			$response_list[ $i ] = $response;
+
+			if ( ! $has_page_cache['server_caching'] &&
+				! empty( $response_list[ $i - 1 ] ) &&
+				! empty( $response_list[ $i ] ) &&
+				$response_list[ $i - 1 ] === $response_list[ $i ]
+			) {
+				$has_page_cache['server_caching'] = true;
+				break;
+			}
+		}
+
+		return $has_page_cache;
+	}
+
+	/**
+	 * Send render random in header if requested.
+	 *
+	 * @return void
+	 */
+	public function maybe_send_random_number() {
+
+		if ( isset( $_SERVER['HTTP_X_AMP_REQUEST_RANDOM_NUMBER'] ) ) {
```

Comment:
Instead of sending the response header whenever the `X-AMP-Request-Random-Number` request header is present, I wonder if it would better to have the client send a nonce instead of `true` as request header value, and then only send the response if the nonce matches the expected. That would ensure the random number is only ever sent back if there was a valid request for it.